nixos/modules/services/torrent/qbittorrent.nix

```diff
@@ -0,0 +1,228 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.qbittorrent;
+  inherit (lib) literalExpression;
+  inherit (lib.types) str unspecified;
+  inherit (lib.meta) getExe maintainers;
+  inherit (lib.options) mkEnableOption mkOption mkPackageOption;
+  inherit (lib.modules) mkIf;
+
+  inherit (builtins) concatStringsSep isAttrs;
+  inherit (lib.attrsets) mapAttrsToList;
+  inherit (lib.strings) escape;
+  inherit (lib.generators) toINI mkKeyValueDefault mkValueStringDefault;
+  gendeepINI = toINI {
+    mkKeyValue =
+      let
+        sep = "=";
+      in
+      k: v:
+      if isAttrs v then
+        concatStringsSep "\n" (
+          mapAttrsToList (k2: v2: "${escape [ sep ] "${k}\\${k2}"}${sep}${mkValueStringDefault { } v2}") v
+        )
+      else
+        mkKeyValueDefault { } sep k v;
+  };
+in
+{
+  options.services.qbittorrent = {
+    enable = mkEnableOption "qbittorrent, BitTorrent client.";
+
+    package = mkPackageOption pkgs "qbittorrent-nox" { };
+
+    openFirewall = mkEnableOption "Opens the default webport and if defined the serverConfig's set ports.";
+
+    user = mkOption {
+      type = str;
+      default = "qbittorrent";
+      description = "User account under which qbittorrent runs.";
+    };
+
+    group = mkOption {
+      type = str;
+      default = "qbittorrent";
+      description = "Group under which qbittorrent runs.";
+    };
+
+    profileDir = mkOption {
+      type = str;
+      default = "/var/lib/qBittorrent/";
+      description = "the path passed to qbittorrent via --profile.";
+    };
+
+    serverConfig = mkOption {
+      description = ''
+        Free-form settings mapped to the `qBittorrent.conf` file in the profile.
+        Refer to <https://github.com/qbittorrent/qBittorrent/wiki/Explanation-of-Options-in-qBittorrent>
+        you will probably want to run qBittorrent locally once to use the webui to generate the password ByteArray before setting anything here.
+      '';
+      example = literalExpression ''
+        {
+          LegalNotice.Accepted = true;
+          BitTorrent.Session = {
+            port = 43862;
+          };
+          Prefences = {
+            WebUI = {
+              Port = 8077;
+              Username = "user";
+              Passowrd_PBKDR2 = "generated ByteArray.";
```

Comment:
So far, these settings have to be provided.
The way I've created this password so far is by running the command manually once, copy the generated password, change the password in the UI, then re-run the service, which is not ideal.
I'll keep looking

Reply:
I've looked into generating the password and it seems like a massive pain, would require some script or whatever else, will add this to the would like to haves

Reply:
Sounds good, let me give it another try today

Reply:
I have a working solution on my profile linked above for reference, of course paths and password will need to changed accordingly etc etc.
also feel free to reach out over matrix either personally or in the nix space. (@nuko:nuko.city)